### PR TITLE
feat(KIT-0049): shape-scoped project sync — manifest intersection replaces the refusal

### PR DESCRIPTION
**Pull request by movito**
## Summary

Replaces the KIT-0048 refusal stopgap: planning-shaped repos now sync by **manifest intersection**. Both locked decisions honored — single source = the consumer's own `scripts/.core-manifest.json`; trigger = implicit shape-read via `_doctor_shape()`, no flag.

**The one design call** (stated per handoff): the intersection lives in the **engine** (`SyncOptions.allowlist`), not the wrapper — because manifest preservation has to happen inside `_build_new_manifest`, which only the engine reaches. The wrapper's job is reading the shape + supplying the recorded entry list.

- **Engine**: `allowlist` intersects candidates after `--tier`/`--only` narrowing; excluded upstream entries land in `SyncReport.skipped_additions` — **named, never silent, deliberately NOT warnings** (exit contract 0/1/2/3/4 untouched). `complete` = "synced everything the allowlist records within entitlement", so `core_version` still converges. Preserved manifests are **pruned of entries upstream no longer ships** (evaluator catch — a verbatim copy would freeze stale records forever). `--only` naming an entry outside the allowlist is a UsageError, never a silent drop.
- **Wrapper**: planning shape loads the consumer manifest → allowlist (validated against the source-manifest path rules — traversal/absolute entries refuse); malformed shape still refuses exit 2; missing/corrupt/null-files manifest refuses exit 2; skipped additions printed with the re-bootstrap pointer.
- **v1 limitation, loud**: planning repos receive *updates* to recorded files, not *additions* — skipped additions are named in every report; additions arrive via re-bootstrap or a manual manifest add (P3-era revisit recorded in the spec).

## Evaluator review (trio BEFORE PR)

fast-v2 CONCERNS / **o3 FAIL** / claude-code CONCERNS → **7 accepted** (headline: the upstream-deletion divergence — preserve-verbatim froze stale records while core_version converged; also the `--only`/allowlist silent drop, a stranded `skipif` decorator pytest silently ignored [consumer CI would error, not skip], `"files": null` AttributeError, traversal-entry defense-in-depth), **4 declined with evidence**. Disposition: `.kit/context/reviews/KIT-0049-evaluator-review.md`.

## Test plan

- [x] 31 tests in `tests/test_project_sync.py` — planning updates-only-recorded, reduced-manifest preservation pin, skipped-addition naming, **single-shape characterization (additions still arrive, full manifest replaced — byte-identical behavior)**, upstream-deletion pruning, `--only`-clash exit 2, malformed-shape / missing-manifest / null-files / traversal refusals, guard pass-through, engine-level report fields
- [x] `ci-check.sh` green; canary green after every pre-commit run
- [x] Engine exit-code contract untouched (frozen-contract tests still pass)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01TULQcU9dmD1fxxZNZ6fD1a

---

> [!NOTE]
> **Medium Risk**
> Changes core sync engine semantics (completeness, manifest writes, and reporting) and the planning provisioning path; behavior is heavily tested and exit codes stay frozen, but mistakes could mis-sync or stall `core_version` for planning consumers.
> 
> **Overview**
> **Planning-shaped repos can sync again** — the KIT-0048 hard refusal is removed and replaced with **manifest-intersection** sync driven by the consumer’s own `scripts/.core-manifest.json`.
> 
> The **sync engine** gains `SyncOptions.allowlist`: after tier/`--only` narrowing, candidates are intersected with the allowlist; upstream entries outside it are reported in **`SyncReport.skipped_additions`** (named in CLI output, not warnings). Allowlist runs **preserve the target’s reduced `files` manifest** (with **pruning** when upstream drops entries), redefine **`complete`** as “all allowlisted entries within entitlement” (blocking vacuous bumps), and treat **`--only` outside the allowlist** as a `UsageError`.
> 
> The **`project sync` wrapper** loads and validates the planning consumer manifest into the allowlist (schema, unsafe paths, empty list) and passes it through on dry-run and apply; malformed shape still exits **2**.
> 
> **Tests** replace the old shape-guard class with a broad `TestShapeScopedSync` suite (planning updates-only, manifest pin, skipped-addition messaging, single-shape parity, refusal edge cases, engine report fields). Task/review kit docs move KIT-0049 to in-review.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f23fb6fecddd0ca8ce07bea1474113cd69beb670. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/core/project [hunk 1]
```diff
@@ -1534,15 +1549,59 @@ def cmd_sync(args, project_dir):
...
+            with open(manifest_path, encoding="utf-8") as f:
+                consumer_manifest = json.load(f)
+            files_section = consumer_manifest["files"]
+            allowlist = sorted(
+                {
+                    entry
+                    for entries in files_section.values()
+                    for entry in entries
+                    if isinstance(entry, str)
+                }
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/core/project]: Fixed in 717c6cd — tiers are validated as lists before the allowlist is built; a string-valued tier now refuses with exit 2 (schema violation) instead of char-iterating into a poisoned allowlist. Regression test added.

File: scripts/core/project [hunk 2]
```diff
@@ -1534,15 +1549,59 @@ def cmd_sync(args, project_dir):
...
+            )
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/core/project:1581]: Fixed across 717c6cd + 3b3fbd2 — files is now validated as dict[str, list[str]] before the allowlist is built: string-valued tiers (would char-iterate) and non-string list members (were silently discarded) both refuse with exit 2. Regression tests for both shapes of the violation (33 total).

File: scripts/core/sync_from_manifest.py [hunk 3]
```diff
@@ -669,8 +709,42 @@ def sync(source: str | Path, target: str | Path, options: SyncOptions) -> SyncRe
...
+        )
+        candidates = [(tier, entry) for tier, entry in candidates if entry in allow]
+
     effective_entries = {entry for _tier, entry in candidates}
-    complete = effective_entries == full_entitlement
+    if options.allowlist is not None:
+        # complete = "synced everything the allowlist records within this
+        # target's entitlement" — an allowlisted target that got all its
+        # recorded files IS complete (core_version bumps, no partial flag).
+        complete = effective_entries == (full_entitlement & set(options.allowlist))
```
[the remark this answers is not in the document]
Comment (movito) [reply scripts/core/sync_from_manifest.py]: Fixed in 4e7bf4a — an empty entitlement∩allowlist scope is never complete: nothing was verified against upstream, so the vacuous match no longer advances core_version. Regression test records only an opt-gated entry without the opt-in and asserts complete=False with core_version untouched.
